src/templates/paint_on_template_tool.cpp

```diff
@@ -195,16 +195,8 @@ ActionGridBar* PaintOnTemplateTool::makeToolBar()
 	auto* toolbar = new ActionGridBar(ActionGridBar::Horizontal, 2);
 	auto* color_options = new QActionGroup(this);
 	
-	int count = 0;
-	static QColor const default_colors[] = {
-	    qRgb(255,   0,   0),
-	    qRgb(255, 255,   0),
-	    qRgb(  0, 255,   0),
-	    qRgb(219,   0, 216),
-	    qRgb(  0,   0, 255),
-	    qRgb(209,  92,   0),
-	    qRgb(  0,   0,   0),
-	};
+	auto const default_colors = Settings::getInstance().paintOnTemplateColors();
+	int count = !(default_colors.size() % 2); // align colors with the other icons
```

Comment:
Looks strange to me with type `int` and operator `!`. Do you mean:
~~~
int count = (default_colors.size() + 1) % 2;
~~~

Reply:
The combination of `int` and boolean NOT is a common C idiom. But we are typing C++, so I've adopted your expression. :)